File: typescript/huifier_test.ts
import { Huifier } from './huifier';

 describe('Huifier', function () {

  let huifier = new Huifier();

  it('should huify words', function () {
    expect(huifier.huify('Вилка')).toEqual('Хуилка');
    expect(huifier.huify('Толик')).toEqual('Хуелик');
    expect(huifier.huify('Яблоко')).toEqual('Хуяблоко');
    expect(huifier.huify('Красный')).toEqual('Хуясный');
    expect(huifier.huify('Круассан')).toEqual('Хуюассан');
    expect(huifier.huify('Хрю')).toEqual('Хую');
  });

  it('should apply rules for vowels', function () {
    expect(huifier.huify('Арка')).toEqual('Хуярка');
    expect(huifier.huify('Едут')).toEqual('Хуедут');
    expect(huifier.huify('Ёжик')).toEqual('Хуёжик');
    expect(huifier.huify('Иглу')).toEqual('Хуиглу');
    expect(huifier.huify('Окна')).toEqual('Хуекна');
    expect(huifier.huify('Урна')).toEqual('Хуюрна');
    expect(huifier.huify('Стык')).toEqual('Хуик');
    expect(huifier.huify('Эпос')).toEqual('Хуепос');
    expect(huifier.huify('Юрта')).toEqual('Хуюрта');
    expect(huifier.huify('Ящик')).toEqual('Хуящик');
  });

  it('should handle different letter cases', function () {
    expect(huifier.huify('Вилка')).toEqual('Хуилка');
    expect(huifier.huify('вилка')).toEqual('Хуилка');
    expect(huifier.huify('вилка')).toEqual('Хуилка');
    expect(huifier.huify('вИлКа')).toEqual('Хуилка');
  });

  it('should normalize the input', function () {
    expect(huifier.huify('')).toEqual('Хуй');
    expect(huifier.huify(' ')).toEqual('Хуй');
    expect(huifier.huify('   ')).toEqual('Хуй');
    expect(huifier.huify('  Вилка  ')).toEqual('Хуилка');
    expect(huifier.huify('Й')).toEqual('Хуй');
    expect(huifier.huify('А')).toEqual('Хуя');
  });
});

File: dart/huifier_test.dart
import 'huifier.dart';
import 'package:test/test.dart';

void main() {
  test('Should huify words', () {
    expect(huify('Вилка'),'Хуилка');
    expect(huify('Толик'),'Хуелик');
    expect(huify('Яблоко'),'Хуяблоко');
    expect(huify('Красный'),'Хуясный');
    expect(huify('Круассан'),'Хуюассан');
    expect(huify('Хрю'),'Хую');
  });

  test('Should apply rules for vowels', () {
    expect(huify('Арка'),'Хуярка');
    expect(huify('Едут'),'Хуедут');
    expect(huify('Ёжик'),'Хуёжик');
    expect(huify('Иглу'),'Хуиглу');
    expect(huify('Окна'),'Хуекна');
    expect(huify('Урна'),'Хуюрна');
    expect(huify('Стык'),'Хуик');
    expect(huify('Эпос'),'Хуепос');
    expect(huify('Юрта'),'Хуюрта');
    expect(huify('Ящик'),'Хуящик');
  });

  test('Should handle different letter cases', () {
    expect(huify('Вилка'),'Хуилка');
    expect(huify('вилка'),'Хуилка');
    expect(huify('вилка'),'Хуилка');
    expect(huify('вИлКа'),'Хуилка');
  });

  test('Should normalize the input', () {
    expect(huify(''),'Хуй');
    expect(huify(' '),'Хуй');
    expect(huify('   '),'Хуй');
    expect(huify('  Вилка  '),'Хуилка');
    expect(huify('Й'),'Хуй');
    expect(huify('А'),'Хуя');
  });
}
